### ISSUE TITLE
Corrupted savefile for Angband 4.2.1??

### ISSUE THREAD

**Issue by RevJim1969** [angband/angband#4557]
I saved and exited the program no problem. When I try to open my save file, I get the following error message:
Assertion failed!
Program:...
File: load.c
Line: 334
Expression: c->objects[obj->oidx] == NULL

I tried to upload the save file, but I get a message saying "We don't support that file type"

Anything you can do to assist would be greatly appreciated.

**Comment by NickMcConnell** [comment]
Where are you trying to upload the file to?  Sometimes compressing it to a .zip file will do the trick.

**Comment by RevJim1969** [comment]
Here is the file in .zip form. Thanks!!
[Creed_Bratton.zip](https://github.com/angband/angband/files/5322510/Creed_Bratton.zip)

**Comment by NickMcConnell** [comment]
Sorry, I'll also need your randart file.  It should be in your lib\user\archive directory and be called randart_04a1c39a.txt.

**Comment by RevJim1969** [comment]
Here it is. Thanks again!
[randart_04a1c39a.txt](https://github.com/angband/angband/files/5322560/randart_04a1c39a.txt)

**Comment by NickMcConnell** [comment]
OK, try this: 
[Creed_Bratton.zip](https://github.com/angband/angband/files/5322628/Creed_Bratton.zip)

**Comment by RevJim1969** [comment]
Yes it worked! Thank you so very much, I had been playing this game for almost two weeks. In layman's terms, what happened? Thanks again!

**Comment by NickMcConnell** [comment]
There's a thing that happens (which I don't fully understand) where a mimicking monster can be saved without its mimicked object being saved correctly.  I had made a workaround of creating a new object while loading the file, but this workaround was causing the crash that you were getting.  Fixing your file was not too hard, but thanks to you I have now also done a better workaround (namely just deleting the mimic if if can't find its object).

**Comment by RevJim1969** [comment]
Interesting - thanks for explaining to me and again, thank you for saving a character/game that has been almost two weeks in the works!
